packages/web/app/src/components/target/history/errors-and-changes.tsx

```diff
@@ -32,20 +34,97 @@ export function ChangesBlock(props: {
   changes: SchemaChangeFieldsFragment[];
   criticality: CriticalityLevel;
 }): ReactElement | null {
+  const router = useRouteSelector();
+
   return (
     <div>
       <h2 className="mb-2 text-sm font-medium text-gray-900 dark:text-white">
         {titleMap[props.criticality]}
       </h2>
-      <ul className="list-inside list-disc pl-3 text-sm leading-relaxed">
+      <ul className="list-inside list-disc space-y-2 pl-3 text-sm leading-relaxed">
         {props.changes.map((change, key) => (
           <li
             key={key}
-            className={clsx(criticalityLevelMapping[props.criticality] ?? 'text-red-400', ' my-1')}
+            className={clsx(
+              criticalityLevelMapping[props.criticality] ?? 'text-red-400',
+              ' my-1 flex space-x-2',
+            )}
           >
             <MaybeWrapTooltip tooltip={change.criticalityReason ?? null}>
               <span className="text-gray-600 dark:text-white">{labelize(change.message)}</span>
             </MaybeWrapTooltip>
+            {change.affectedOperations?.length ? (
+              <Tooltip.Provider delayDuration={0}>
+                <Tooltip
+                  content={
+                    <>
+                      <div className="text-lg font-bold">Usage</div>
+                      <table className="mt-2 table-auto">
+                        <thead>
+                          <tr>
+                            <th className="p-2 pl-0 text-left">Top 10 Operations</th>
+                            <th className="p-2 text-center">Reqs (last 7 days)</th>
+                          </tr>
+                        </thead>
+                        <tbody>
+                          {change.affectedOperations.map(({ hash, name, count }) => (
+                            <tr key={hash}>
+                              <td className="px-2 pl-0 text-left">
+                                <Link
+                                  className="text-orange-500 hover:text-orange-500 hover:underline hover:underline-offset-2"
+                                  href={{
+                                    pathname:
+                                      '/[organizationId]/[projectId]/[targetId]/insights/[operationName]/[operationHash]',
+                                    query: {
+                                      organizationId: router.organizationId,
+                                      projectId: router.projectId,
+                                      targetId: router.targetId,
+                                      operationName: `${hash.substring(0, 4)}_${name}`,
+                                      operationHash: hash,
+                                      period: '7d',
+                                    },
+                                  }}
+                                >
+                                  {hash.substring(0, 4)}_{name}
+                                </Link>
```

Comment:
Question: Currently, if we open the insights page with a non-existing operation, it just shows empty stats components. Should we change it to display some information that _we don't have any stats on this operation/  operation does not exist_?

Reply:
@beerose That is a good idea!

Reply:
I figured that can be a separate PR as it will require more changes